### ISSUE TITLE
ConflictingRegionsException when invoking formatter

### ISSUE THREAD

**Issue by sjiherzig** [JPL-IMCE/gov.nasa.jpl.imce.oml#167]
Since upgrading from M09 to M20, I get the following stack trace:

```
0    [main] ERROR clipse.xtext.util.ExceptionAcceptor  - Conflicting Regions.
{{{}}}: newLine=1 at offset=29 length=0
[[[]]]: newLine=2;indentInc=1 at offset=29 length=0
------------------------------- document snippet -------------------------------
openterminology<http://test>{{{{}}}[[[]]]conceptConcept1}
--------------------------------------------------------------------------------
org.eclipse.xtext.formatting2.internal.ConflictingRegionsException: Conflicting Regions.
{{{}}}: newLine=1 at offset=29 length=0
[[[]]]: newLine=2;indentInc=1 at offset=29 length=0
------------------------------- document snippet -------------------------------
openterminology<http://test>{{{{}}}[[[]]]conceptConcept1}
--------------------------------------------------------------------------------
	at org.eclipse.xtext.formatting2.internal.TextSegmentSet.handleConflict(TextSegmentSet.java:83)
	at org.eclipse.xtext.formatting2.internal.ArrayListTextSegmentSet.replaceExistingEntry(ArrayListTextSegmentSet.java:139)
	at org.eclipse.xtext.formatting2.internal.ArrayListTextSegmentSet.add(ArrayListTextSegmentSet.java:49)
	at org.eclipse.xtext.formatting2.internal.FormattableDocument.addReplacer(FormattableDocument.java:77)
	at org.eclipse.xtext.formatting2.internal.FormattableDocument.set(FormattableDocument.java:326)
	at org.eclipse.xtext.formatting2.internal.FormattableDocument.prepend(FormattableDocument.java:278)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter._format(OMLFormatter.java:414)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter.format(OMLFormatter.java:2277)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter.lambda$15(OMLFormatter.java:211)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter._format(OMLFormatter.java:217)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter.format(OMLFormatter.java:2340)
	at org.eclipse.xtext.formatting2.internal.FormattableDocument.format(FormattableDocument.java:194)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter.lambda$2(OMLFormatter.java:140)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter._format(OMLFormatter.java:142)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter.format(OMLFormatter.java:2433)
	at org.eclipse.xtext.formatting2.internal.FormattableDocument.format(FormattableDocument.java:194)
	at org.eclipse.xtext.formatting2.AbstractFormatter2._format(AbstractFormatter2.java:203)
	at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter.format(OMLFormatter.java:2424)
	at org.eclipse.xtext.formatting2.AbstractFormatter2.handleTraceMissing(AbstractFormatter2.java:290)
	at org.eclipse.xtext.formatting2.AbstractFormatter2.format(AbstractFormatter2.java:273)
	at org.eclipse.xtext.serializer.impl.Serializer.serialize(Serializer.java:147)
	at org.eclipse.xtext.serializer.impl.Serializer.serialize(Serializer.java:189)
	at org.eclipse.xtext.resource.XtextResource.doSave(XtextResource.java:386)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.save(ResourceImpl.java:1430)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.save(ResourceImpl.java:999)
	at gov.nasa.jpl.imce.oml.dsl.tests.OMLAPITest1.main(OMLAPITest1.java:36)
	Suppressed: org.eclipse.xtext.formatting2.internal.RegionTrace: newLine=1
		at org.eclipse.xtext.formatting2.internal.TextSegmentSet.trace(TextSegmentSet.java:68)
		at org.eclipse.xtext.formatting2.internal.ArrayListTextSegmentSet.add(ArrayListTextSegmentSet.java:43)
		... 25 more
	Suppressed: org.eclipse.xtext.formatting2.internal.RegionTrace: newLine=2;indentInc=1
		at org.eclipse.xtext.formatting2.internal.TextSegmentSet.trace(TextSegmentSet.java:68)
		at org.eclipse.xtext.formatting2.internal.ArrayListTextSegmentSet.replaceExistingEntry(ArrayListTextSegmentSet.java:130)
		at org.eclipse.xtext.formatting2.internal.ArrayListTextSegmentSet.add(ArrayListTextSegmentSet.java:49)
		at org.eclipse.xtext.formatting2.internal.FormattableDocument.addReplacer(FormattableDocument.java:77)
		at org.eclipse.xtext.formatting2.internal.FormattableDocument.set(FormattableDocument.java:313)
		at org.eclipse.xtext.formatting2.internal.FormattableDocument.interior(FormattableDocument.java:252)
		at gov.nasa.jpl.imce.oml.dsl.formatting2.OMLFormatter._format(OMLFormatter.java:193)
		... 17 more
Caused by: org.eclipse.xtext.formatting2.ConflictingFormattingException: Conflicting values for 'newLineMin': '2' and '1'.
	at org.eclipse.xtext.formatting2.internal.HiddenRegionFormatting.merge(HiddenRegionFormatting.java:102)
	at org.eclipse.xtext.formatting2.internal.HiddenRegionFormatting.mergeValuesFrom(HiddenRegionFormatting.java:111)
	at org.eclipse.xtext.formatting2.internal.HiddenRegionFormattingMerger.merge(HiddenRegionFormattingMerger.java:32)
	at org.eclipse.xtext.formatting2.internal.HiddenRegionFormattingMerger.merge(HiddenRegionFormattingMerger.java:19)
	at org.eclipse.xtext.formatting2.internal.TextReplacerMerger.mergeHiddenRegionReplacers(TextReplacerMerger.java:73)
	at org.eclipse.xtext.formatting2.internal.TextReplacerMerger.merge(TextReplacerMerger.java:36)
	at org.eclipse.xtext.formatting2.internal.TextReplacerMerger.merge(TextReplacerMerger.java:23)
	at org.eclipse.xtext.formatting2.internal.ArrayListTextSegmentSet.replaceExistingEntry(ArrayListTextSegmentSet.java:128)
	... 26 more
```

This seems to happen only with TerminologyBoxes that are being created. The following is a simple example that reproduces the problem:

```xtend
val extent1 = CommonFactory.eINSTANCE.createExtent
		
val tbox1 = GraphsFactory.eINSTANCE.createTerminologyGraph
tbox1.extent = extent1
tbox1.iri = "http://test"
		
val concept1 = TerminologiesFactory.eINSTANCE.createConcept
concept1.name = "Concept1"
concept1.tbox = tbox1
```

After calling `save(...)` on the `Resource`, the stack trace appears.

**Comment by sjiherzig** [comment]
While the proposed fix works in some cases, the issue prevails in others. The following example still triggers the exception:

```
XcoreStandaloneSetup.doSetup
OMLStandaloneSetup.doSetup

val omlCommonFactory = CommonFactory.eINSTANCE
val omlTerminologiesFactory = TerminologiesFactory.eINSTANCE
val omlGraphsFactory = GraphsFactory.eINSTANCE

val extent1 = omlCommonFactory.createExtent
val extent2 = omlCommonFactory.createExtent

/** TBOX 1 */
val tbox1 = omlGraphsFactory.createTerminologyGraph
tbox1.extent = extent1
tbox1.iri = "http://test"

val concept1 = omlTerminologiesFactory.createConcept
concept1.name = "Concept1"
concept1.tbox = tbox1

/** TBOX 2 */
val tbox2 = omlGraphsFactory.createTerminologyGraph
tbox2.extent = extent2
tbox2.iri = "http://test2"

val terminologyExtension1 = omlTerminologiesFactory.createTerminologyExtensionAxiom
terminologyExtension1.extendedTerminology = tbox1
terminologyExtension1.tbox = tbox2

val concept2 = omlTerminologiesFactory.createConcept
concept2.name = "Concept2"
concept2.tbox = tbox2

val concept2Extends = omlTerminologiesFactory.createConceptSpecializationAxiom
concept2Extends.superConcept = concept1
concept2Extends.subConcept = concept2
concept2Extends.tbox = tbox2

val resourceSet = new XtextResourceSet()

val res = resourceSet.createResource(URI.createURI("test.oml"))
res.getContents().add(extent1)
res.save(Collections.EMPTY_MAP)

val res2 = resourceSet.createResource(URI.createURI("test2.oml"))
res2.getContents().add(extent2)
res2.save(Collections.EMPTY_MAP)
```

It seems that, in this specific case, this is tied to extending graphs.

**Comment by NicolasRouquette** [comment]
Added a new unit test: /gov.nasa.jpl.imce.oml.dsl.tests/src/gov/nasa/jpl/imce/oml/dsl/tests/OMLFormattingTest3.xtend

This works with release 0.9.0.0-M21

**Comment by NicolasRouquette** [comment]
This was a fluke; the fixes to #172 were not in M21 yet.
